src/protocol_level_tokens/token_client.rs

```diff
@@ -103,8 +99,18 @@ pub enum TokenError {
     /// burn the total amount in the payload.
     #[error("Total token amount in the payload exceeds governance account balance.")]
     InsufficientSupply,
-    #[error("The token state is paused")]
+    #[error("The token state is paused.")]
     Paused,
+    /// Error that indicates that the token does not support minting.
+    #[error("The token cannot be minted.")]
+    NotMintable,
+    /// Error that indicates that the token does not support burning.
+    #[error("The token cannot be burned.")]
+    NotBurnable,
+    /// Error that indicates that allow/deny list is not available for this
+    /// token.
+    #[error("The token does not have allow/deny list")]
+    NoList,
```

Comment:
Suggestion to have two different errors for each list check.

```suggestion
    #[error("The token does not have allow list")]
    NoAllowList,
    #[error("The token does not have deny list")]
    NoDenyList,
```